examples/python/xds/server.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Python implementation of the GRPC helloworld.Greeter server."""
+
+from concurrent import futures
+import argparse
+import logging
+import socket
+
+import grpc
+
+import helloworld_pb2
+import helloworld_pb2_grpc
+
+from grpc_reflection.v1alpha import reflection
+from grpc_health.v1 import health
+from grpc_health.v1 import health_pb2
+from grpc_health.v1 import health_pb2_grpc
+
+_DESCRIPTION = "A general purpose dummy server."
+
+
+class Greeter(helloworld_pb2_grpc.GreeterServicer):
+
+    def __init__(self, hostname):
+        self._hostname = hostname if hostname else socket.gethostname()
+
+    def SayHello(self, request, context):
+        return helloworld_pb2.HelloReply(
+            message=f"Hello {request.name} from {self._hostname}!")
+
+
+def serve(port, hostname):
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
```

Comment:
What's our take on the `max_workers` variable?

Reply:
I used a (mostly) arbitrary number here. If we really care about the performance, we could do a multiple of `multiprocessing.cpu_count()`, but I don't think it's particularly important here.

Reply:
Switched main threadpool to `multiprocessing.cpu_count()` and health pool to 1.